src/session_pool/session_pool.js

```diff
@@ -168,26 +169,58 @@ export class SessionPool extends EventEmitter {
         events.on(ACTOR_EVENT_NAMES_EX.PERSIST_STATE, this._listener);
     }
 
+    /**
+     * Explicitly creates and adds new session to the session pool.
+     * @param {SessionOptions} [options] - The configuration options for the session being added to the session pool.
+     */
+    async addSession(options = {}) {
+        const { id } = options;
+        if (id) {
+            const sessionExists = this.sessionMap.has(id);
+            if (sessionExists) {
+                throw new Error(`Cannot add session with id '${id}' as it already exists in the pool`);
+            }
+        }
+
+        if (!this._hasSpaceForSession()) {
+            this._removeRetiredSessions();
+        }
+
+        const newSession = await this.createSessionFunction(this, { sessionOptions: options });
+        this.log.debug(`Created new Session - ${newSession.id}`);
+
+        this._addSession(newSession);
+    }
+
     /**
      * Gets session.
-     * If there is space for new session, it creates and return new session.
+     * If there is space for new session, it creates and returns new session.
      * If the session pool is full, it picks a session from the pool,
      * If the picked session is usable it is returned, otherwise it creates and returns a new one.
      *
+     * @param {String} [sessionId] - If provided, it attempts to return the session with corresponding `id` first.
      * @return {Promise<Session>}
      */
-    async getSession() {
+    async getSession(sessionId) {
+        if (sessionId) {
+            const sessionExists = this.sessionMap.has(sessionId);
+            if (sessionExists) {
+                const session = this.sessionMap.get(sessionId);
+                if (session.isUsable()) return session;
+                this._removeRetiredSessions();
```

Comment:
I don't think we need to remove sessions here. We did not create any new session. We remove sessions only before creating new ones, no?

Reply:
Makes sense. Was thinking that maybe it's a good idea to remove the retired session whenever we found it. But yeah - maybe we would not even create new ones so we don't really need it. Fixed.